client/src/glossary/GlossarySidebar/GlossarySidebar.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+
+import "./GlossarySidebar.scss";
+
+import {
+  Sidebar,
+  DebouncedTextInput,
+  create_text_maker_component,
+} from "src/components/index";
+
+import glossary_text from "src/glossary/glossary.yaml";
+
+import { IconSearch } from "src/icons/icons";
+
+import { GlossaryDef } from "./GlossarySidebarDefinition";
+import { GlossaryList } from "./GlossarySidebarList";
+
+const { text_maker } = create_text_maker_component(glossary_text);
+
+const GlossarySidebarSearchDefaultProps = {
+  placeholder: text_maker("glossary_placeholder"),
+};
+
+type GlossarySidebarProps = typeof GlossarySidebarSearchDefaultProps & {
+  glossary_item_key: string;
+  focus_item_key: string;
+  set_glossary_item: (key: string) => void;
+  search_phrase: string;
+  set_query: (query: string) => void;
+  is_open: boolean;
+  toggle_glossary: (value: boolean) => void;
+  return_focus_target: HTMLElement | null;
+};
+
+export class GlossarySidebar extends React.Component<GlossarySidebarProps> {
+  static defaultProps = GlossarySidebarSearchDefaultProps;
+
+  constructor(props: GlossarySidebarProps) {
+    super(props);
+  }
+
+  callback = (query: string) => {
+    this.props.set_query(query);
+    this.props.set_glossary_item("");
+  };
+
+  render() {
+    const {
+      glossary_item_key,
+      search_phrase,
+      focus_item_key,
+      placeholder,
+      is_open,
+      return_focus_target,
+      set_glossary_item,
+      toggle_glossary,
+    } = this.props;
+    return (
+      <Sidebar
+        is_open={is_open}
+        open_close_callback={toggle_glossary}
+        title_text={text_maker("glossary_title")}
+        sidebar_toggle_target={"[data-toggle=glossary_sidebar]"}
+        return_focus_target={return_focus_target}
+      >
+        <div className="glossary-sb__header-wrapper">
+          <div className="glossary-sb__header">
+            <div className="glossary-sb__search-wrapper">
+              <div className={"glossary-sb__search-bar"}>
+                <div className={"glossary-sb__icon-container"}>
+                  <span aria-hidden="true">
+                    <IconSearch
+                      width="30px"
+                      color="#2C70C9"
```

Comment:
Should import and use the `secondaryColor` variable, since that appears to be where this is from.